File: 2022_summer/2696.java
import java.util.*;

public class Main {
    public static void main (String[] args){
        Scanner sc = new Scanner(System.in);
        Integer tc;
        tc = sc.nextInt();
        while(true){
            if(tc-- == 0) break;
            PriorityQueue<Integer> lpq = new PriorityQueue<>(Collections.reverseOrder());
            PriorityQueue<Integer> rpq = new PriorityQueue<>();
            ArrayList<Integer> mid = new ArrayList<>();
            Integer n,a,b,pivot;
            n = sc.nextInt();
            pivot = sc.nextInt();
            mid.add(pivot);

            for(int i=2;i<=n;i+=2){
                a = sc.nextInt();
                b = sc.nextInt();

                if(a >=pivot) rpq.add(a);
                else lpq.add(a);

                if(b >= pivot) rpq.add(b);
                else lpq.add(b);

                if(lpq.size() != rpq.size()){
                    if(lpq.size() > rpq.size()){
                        rpq.add(pivot);
                        pivot = lpq.poll();
                    }
                    else{
                        lpq.add(pivot);
                        pivot = rpq.poll();
                    }
                }
                mid.add(pivot);
            }
            System.out.println(mid.size());
            for (int i = 0; i < mid.size(); i++) {
                System.out.print(mid.get(i) + " ");
                if (i % 10 == 9 ) System.out.println();
            }
            System.out.println();
        }
    }
}

File: 2022_summer/2696.cpp
#include <iostream>
#include <queue>
using namespace std;

int tc;

int main() {
	cin >> tc;
	while (tc--) {
		vector<int> mid;
		int n,pivot,a,b;
		
		priority_queue <int, vector<int>, greater<int>> rpq;
		priority_queue<int> lpq;

		cin >> n >> pivot; mid.push_back(pivot);
		for (int i = 2; i <= n; i+=2) {
			cin >> a >> b;
			if (a >= pivot)	rpq.push(a);
			else lpq.push(a);

			if (b >= pivot) rpq.push(b);
			else lpq.push(b);
			
			if (lpq.size() != rpq.size()) {
				if (lpq.size() > rpq.size()) {
					rpq.push(pivot);
					pivot = lpq.top();
					lpq.pop();
				}
				else {
					lpq.push(pivot);
					pivot = rpq.top();
					rpq.pop();
				}
			}
			mid.push_back(pivot);			
		}

		cout << mid.size() << "\n";
		for (int i = 0; i < mid.size(); i++) {
			cout << mid[i] << " ";
			if (i % 10 == 9 ) cout << "\n";
		}

	}

}
